app/src/app/core/services/elasticsearch/data.service.ts

```diff
@@ -217,14 +217,13 @@ export class DataService {
    */
   private async performQuery<T>(query: QueryBuilder, url: string = this.baseUrl, type?: EntityType) {
     const response = await this.http.post<T>(url, query.build()).toPromise();
-    const entities = this.filterData<T>(response, type);
+    const entities = await this.filterData<T>(response, type);
```

Comment:
There might be some problems regarding performquery (you still get the data back in findById)). Most of the pages like location are not shown (cannot find property of  'id' of null). I assume it is performquery since findArtworksByType is invoking this method.

Reply:
I think filterData missed some async calls so replaced it by an hopefully more intuitive forEach functuon. Can you have another look into it?